Added Distance /controllers/index

agbolton opened the pull request:
The distance can be called after getting an array of trails. Trying this simple pull request first. WIll add the other story/subtasks separately. 

Still need to how we are calling these from the front end.

agbolton committed:
Added Distance /controllers/index

Distance can be called after getting an array of trails

agbolton committed:
Ran Prettier

mgarber93 commented on src/controllers/distance.js:
```diff
@@ -0,0 +1,20 @@
+  return distance;
+}
+
+function latLonDistance(trailLat, trailLon, coordinate) {
```
IMO this should have taken two coordinates (start and end?) as parameters. As far as the math goes I'm pretty lost, maybe link an article or something in a comment?

mgarber93 replied:
Maybe use [Polar coordinate flat-Earth formula](https://en.wikipedia.org/wiki/Geographical_distance) with a separate function to convert to colatitude in radians?

agbolton replied:
Yeah just took it straight from here - https://www.geodatasource.com/developers/javascript

agbolton replied:
Also, we definitely can refactor into both being objects and then just referencing .lat / .lon

mgarber93 commented on src/controllers/distance.js:
```diff
@@ -0,0 +1,20 @@
+export function findDistanceToTrail(trail, coordinates) {
```
I don't think this needs to exist.

agbolton committed:
New Function to Take A Zip code and get Lat / Lon

Prob needs its own route - kept it in the trails router.

agbolton committed:
Added MapQuest Key

There is a free tier mapquest key so we can input a Zip code and get the lat/lon

mgarber93 committed:
Merge branch 'main' into distance
